### DISCUSSION TITLE
Remote Upgrade doesn't work for Windows machines?

### DISCUSSION THREAD

**Discussion by bgiesing** [topgrade-rs/topgrade#935] [Q&A]
It connects properly via SSH but then simply says

```
env: The term 'env' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

Any ideas? I know ENV doesn't exist on Windows since that's a Unix command but there doesn't seem to be anyway to bypass it? Only thing changed in my config is actually adding my host name, changing the command to point to the WinGet version (with proper escaping), and uncommenting out those lines, didn't touch the SSH options line at all.

**Comment by SteveLauC** [comment]
Hi, from the [source code](https://github.com/topgrade-rs/topgrade/blob/9767e4169cd392a33417728b189b7cae9551e9e4/src/steps/remote/ssh.rs#L13-L55), the answer seems to be yes. At first glance, 2 Unix things are used here:

1. env

Used to set an environment variable `TOPGRADE_PREFIX={hostname}`

2. `$SHELL`

Used to decide the shell to use

I think we can add Windows support if we replace the above 2 things with Windows counterparts.

**Comment by SteveLauC** [comment]
Created an issue for this: #1027
